api/src/main/java/com/flipkart/poseidon/validator/APIValidator.java

```diff
@@ -42,78 +42,81 @@
     private static final List<EndpointPOJO> pojos = new ArrayList<>();
 
     public static void main(String[] args) {
-        Path dir = Paths.get(args[0]);
-        CustomValidator customValidator = null;
-
-        boolean validateDataSources = false;
-        List<String> packagesToScan = new ArrayList<>();
-
-        if (args.length > 1) {
-            if (!StringUtils.isNullOrEmpty(args[1].trim())) {
-                try {
-                    final Class<?> customValidatorClass = Class.forName(args[1]);
-                    if (!CustomValidator.class.isAssignableFrom(customValidatorClass)) {
-                        logger.error("Validator class passed does not implement CustomValidator");
+        boolean skipAPIValidator = Boolean.valueOf(System.getProperty("skipAPIValidator"));
```

Comment:
API validator is triggered during build only if explicitly specified by the client application using Poseidon. So instead of this change, client app can just remove the validator from its pom (typically invoked via a mvn exec plugin).
But I would suggest to use this validator to avoid runtime errors due to mismatch in api definition file and the actual datasources.